src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/LegacyPbaAccountServiceImpl.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.hmcts.reform.professionalapi.service.impl;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.stereotype.Service;
+
+import uk.gov.hmcts.reform.professionalapi.domain.PaymentAccount;
+import uk.gov.hmcts.reform.professionalapi.domain.ProfessionalUser;
+import uk.gov.hmcts.reform.professionalapi.domain.UserAccountMap;
+import uk.gov.hmcts.reform.professionalapi.service.LegacyPbaAccountService;
+
+@Service
+@Slf4j
+public class LegacyPbaAccountServiceImpl implements LegacyPbaAccountService {
+
+    List<String> paymentAccountPbaNumbers = new ArrayList<>();
+
+    public List<String> findLegacyPbaAccountByUserEmail(ProfessionalUser professionalUser) {
+
+        List<String> pbaNumbers = null;
+        List<PaymentAccount>  paymentAccountsFromEntity = new ArrayList<>();
+
+        if (professionalUser.getOrganisation().getPaymentAccounts().isEmpty()) {

```

Comment:
This is overly complicated. Please change to just be a single if and negate it to:
if (!professionalUser.getOrganisation().getPaymentAccounts().isEmpty()) {
...
}
INSTEAD OF what is currently in else. You can assign pbaNumbers to an ArrayList when they are defined

Reply:
done